### PR TITLE
Uninstall Script

### PR DESCRIPTION
**Pull request by samvictordr**
Added uninstall script to undo all changes done to the system after running install.sh, removes all installed yay packages and their dependencies, and ags.

yes, it removes the user from video/input groups as well.

### CODE REVIEW

File: uninstall.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Undo Step 3: Removing copied config and local folders
+echo 'Removing copied config and local folders...'
+rm -rf "$HOME/.config"
```
Comment (end-4) [inline uninstall.sh]: no no no no :skull:

File: uninstall.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Undo Step 3: Removing copied config and local folders
+echo 'Removing copied config and local folders...'
+rm -rf "$HOME/.config"
+rm -rf "$HOME/.local"
+
+# Undo Step 2: Uninstall AGS
+echo 'Uninstalling AGS...'
+sudo meson uninstall -C ~/ags/build
```
Comment (end-4) [inline uninstall.sh]: ags isn't here though? 
1. the install script is meant to be run from the repo folder, not home
2. ags is actually something in /usr/bin/ags... but that's just the launcher script. i'm not sure how we can actually get rid of the other libraries

File: uninstall.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# Undo Step 1: Remove added user from video and input groups and remove yay packages
+echo 'Removing user from video and input groups and removing packages...'
+user=$(whoami)
+sudo deluser "$user" video
+sudo deluser "$user" input
+echo 'User rmoved from video and input groups.'
+
+# Removing installed yay packages and dependencies
+
+yay -Rns brightnessctl coreutils curl fish foot fuzzel gjs gnome-bluetooth-3.0 gnome-control-center gnome-keyring gobject-introspection grim gtk3 gtk-layer-shell libdbusmenu-gtk3 meson networkmanager nlohmann-json npm plasma-browser-integration playerctl polkit-gnome python-pywal ripgrep sassc slurp starship swayidle swaylock typescript upower xorg-xrandr webp-pixbuf-loader wget wireplumber wl-clipboard tesseract yad ydotool adw-gtk3-git cava gojq gradience-git gtklock gtklock-playerctl-module gtklock-powerbar-module gtklock-userinfo-module hyprland-git lexend-fonts-git python-material-color-utilities python-pywal python-poetry python-build python-pillow swww ttf-material-symbols-variable-git ttf-space-mono-nerd ttf-jetbrains-mono-nerd wlogout
```
Comment (end-4) [inline uninstall.sh]: i'm having doubts... this is kinda destructive
i don't think it's a good idea to remove `coreutils upower ripgrep wget`
also remove nlohmann json please, this is no longer needed but i forgot to remove

**Review by end-4 — CHANGES_REQUESTED** [review]

File: uninstall.sh (hunk 3 above)
Comment (samvictordr) [reply uninstall.sh]: Makes sense, some of the packages are included with a gnome install after all

I'll get back to this after testing it on a system with hyprland installed

File: uninstall.sh (hunk 1 above)
Comment (samvictordr) [reply uninstall.sh]: I really didn't think that through lmao

Comment (samvictordr) [reply uninstall.sh]: Fixed it

File: uninstall.sh (hunk 3 above)
Comment (samvictordr) [reply uninstall.sh]: I've removed arch-crucial packages and nlohmann-json, after comparing the vanilla arch packages installed with the packages to the removed above (I've removed wireplumber from the list to be uninstalled as well as it's usually installed with pipewire)

File: uninstall.sh (hunk 2 above)
Comment (samvictordr) [reply uninstall.sh]: Ah well, I've not had much experience with ags
We could just comment that out for now, and enable it after we get a viable solution to remove ags in it's entirety. Until then, I'll create a issue for this.

**Review by samvictordr — COMMENTED** [review]
Done with two requested fixes, and disabled the ags step until a good solution is found to fully uninstall ags

**Comment by end-4** [comment]
- Don't delete the whole `~/.local/bin`, for the same reason as `~/.config`
- And I almost forgot, the script should print the commands that it runs, like `install.sh`. Yknow, in case that huge uninstall command fails (since it might break their other things), the user knows what packages there are to uninstall manually.

**Comment by samvictordr** [comment]
Updated the script, and included a few touches from the original install script as well

File: uninstall.sh [hunk 4]
```diff
@@ -1,24 +1,44 @@
...
+echo '            3. will show all commands that it runs.'
+echo '            4. should be run from its folder.'
+echo 'Ctrl+C to exit. Enter to continue.'
+read
+
+##############################################################################################################################
+
 # Undo Step 3: Removing copied config and local folders
 echo 'Removing copied config and local folders...'
+echo 'rm -rf "$HOME/.config/ags fish frontconfig foot fuzzel gtklock hypr mpv swaylock wlogout starship.toml" '
```
Comment (end-4) [inline uninstall.sh]: this syntax won't work
remove each folder in one command

File: uninstall.sh [hunk 5]
```diff
@@ -1,24 +1,44 @@
...
 rm -rf "$HOME/.config/ags fish frontconfig foot fuzzel gtklock hypr mpv swaylock wlogout starship.toml"
-rm -rf "$HOME/.local/bin"
+echo 'rm -rf "$HOME/.local/bin/fuzzel-emoji rubyshot" '
+rm -rf "$HOME/.local/bin/fuzzel-emoji rubyshot"
```
Comment (end-4) [inline uninstall.sh]: same here

**Review by end-4 — CHANGES_REQUESTED** [review]

File: uninstall.sh (hunk 4 above)
Comment (end-4) [reply uninstall.sh]: actually, no need to remove each in a command
just have multiple targets in one line but not this way
something like `rm -rf "$HOME/.config/ags" "$HOME/.config/fish"`

File: uninstall.sh (hunk 5 above)
Comment (samvictordr) [reply uninstall.sh]: Fixed with [91dc06d](https://github.com/end-4/dots-hyprland/pull/148/commits/91dc06dd1b7272b0516c86b881ca649a777557d5) as well

File: uninstall.sh (hunk 4 above)
Comment (samvictordr) [reply uninstall.sh]: Fixed with commit [91dc06d](https://github.com/end-4/dots-hyprland/pull/148/commits/91dc06dd1b7272b0516c86b881ca649a777557d5)

**Pull request merged**